app/models/api/v1/report.rb

```diff
@@ -70,18 +71,31 @@ def student_json(student)
     {
       id: student.id,
       name: student.name,
+      first_name: student.first_name,
+      last_name: student.last_name,
       started_offering:  started_offering
     }
   end
 
-  # Helpers that provide student answers grouped by embeddable key:
 
+  def self.update_report_learners(report_learners)
+    report_learners.each do |learner|
+      if learner.last_report.nil? or learner.last_report < MAX_REPORT_LEARNER_AGE.ago
+        learner.last_report = Time.now
+        learner.update_fields()
+      end
+    end
+  end
+
+  # Helpers that provide student answers grouped by embeddable key:
   def get_student_answers
     # Collect all the student answers for given offering.
     student_ids = @students.map { |s| s.id }
-    answers = Report::Learner.where(offering_id: @offering.id, student_id: student_ids).map do |report_learner|
+    report_learners = Report::Learner.where(offering_id: @offering.id, student_id: student_ids)
+    API::V1::Report.update_report_learners(report_learners)
```

Comment:
@knowuh can you describe why this is necessary?  The report_learner should be updated in the background as data is sent to the portal from LARA. Where you running into places where the report_learner was not correctly configured? 
Perhaps the issue is that initially the report_learner is not initialized. It probably isn't fully initialized until some data has been sent by the portal to LARA. So there could be some time when a